packages/instrumentation-openai/src/image-wrappers.ts

```diff
@@ -0,0 +1,565 @@
+        // For text prompts (image generation and editing)
+        if (params?.prompt) {
+          // Simple token estimation: roughly 4 characters per token for English text
+          estimatedPromptTokens += Math.ceil(params.prompt.length / 4);
+        }
+        
+        // For image inputs (editing and variations)
+        if (params?.image) {
+          // For input images, use a base estimation (actual input image token cost varies by fidelity)
+          // OpenAI doesn't specify exact input image token costs, so we use a reasonable estimate
+          estimatedPromptTokens += 272; // Base cost similar to generating a low quality 1024x1024 image
+        }
+        
+        if (estimatedPromptTokens > 0) {
+          attributes[SpanAttributes.LLM_USAGE_PROMPT_TOKENS] = estimatedPromptTokens;
+        }
+        
+        attributes[SpanAttributes.LLM_USAGE_TOTAL_TOKENS] = estimatedPromptTokens + completionTokens;
+      } catch (error) {
+        // If token calculation fails, continue without it
+        attributes[SpanAttributes.LLM_USAGE_TOTAL_TOKENS] = response.data.length;
+      }
+    } else {
+      // When enrichTokens is disabled, just set total = completion tokens
+      attributes[SpanAttributes.LLM_USAGE_TOTAL_TOKENS] = completionTokens;
+    }
+  }
+
+  if (response.data && response.data.length > 0) {
+    const firstImage = response.data[0];
+    
+    // Handle base64 response
+    if (firstImage.b64_json && uploadCallback) {
+      try {
+        const traceId = span.spanContext().traceId;
+        const spanId = span.spanContext().spanId;
+        
+        const imageUrl = await uploadCallback(
+          traceId,
+          spanId,
+          "generated_image.png",
+          firstImage.b64_json
+        );
+        
+        attributes[`${SpanAttributes.LLM_COMPLETIONS}.0.content`] = JSON.stringify([
+          { type: "image_url", image_url: { url: imageUrl } }
+        ]);
+        attributes[`${SpanAttributes.LLM_COMPLETIONS}.0.role`] = "assistant";
+      } catch (error) {
+        console.error("Failed to upload generated image:", error);
+      }
+    }
+    // Handle URL response - fetch and upload to SDK
+    else if (firstImage.url && uploadCallback) {
+      try {
+        const traceId = span.spanContext().traceId;
+        const spanId = span.spanContext().spanId;
+        
+        // Fetch the image from OpenAI URL and convert to base64 (Node.js)
+        const response = await fetch(firstImage.url);
+        const arrayBuffer = await response.arrayBuffer();
+        const buffer = Buffer.from(arrayBuffer);
+        const base64Data = buffer.toString("base64");
+        
+        const uploadedUrl = await uploadCallback(
+          traceId,
+          spanId,
+          "generated_image.png",
+          base64Data
+        );
+        
+        attributes[`${SpanAttributes.LLM_COMPLETIONS}.0.content`] = JSON.stringify([
+          { type: "image_url", image_url: { url: uploadedUrl } }
+        ]);
```

Comment:
wondering if we need this json or just the url... what are we doing in the python sdk in these cases?

Reply:
exactly this